src/utils/compressImages.ts

```diff
@@ -0,0 +1,123 @@
+import { ImageCompressionOptions } from '../lib/Sendbird';
+import pxToNumber from './pxToNumber';
+import { Logger } from '../lib/SendbirdState';
+
+interface CompressImageParams {
+  imageFile: File;
+  compressionRate: number;
+  resizingWidth?: number;
+  resizingHeight?: number;
+}
+
+const compressImage = ({
+  imageFile,
+  compressionRate,
+  resizingWidth,
+  resizingHeight,
+}: CompressImageParams): Promise<File> => {
+  const image = document.createElement('img');
+  return new Promise((resolve, reject) => {
+    image.src = URL.createObjectURL(imageFile);
+    image.onerror = reject;
+    image.onload = () => {
+      URL.revokeObjectURL(image.src);
+      const canvas = document.createElement('canvas');
+
+      const originWidth = image.width;
+      const originHeight = image.height;
+      const widthRatio = originWidth / (resizingWidth || originWidth);
+      const heightRatio = originHeight / (resizingHeight || originHeight);
+
+      let targetResizingWidth = resizingWidth || originWidth;
+      let targetResizingHeight = resizingHeight || originHeight;
+
+      // Use the more impactful value, so the original images' ratio won't be broken
+      if (widthRatio > heightRatio) {
+        targetResizingHeight = originHeight / (resizingWidth ? widthRatio : 1);
+      } else if (heightRatio > widthRatio) {
+        targetResizingWidth = originWidth / (resizingHeight ? heightRatio : 1);
+      }
```

Comment:
It would be good to extract the part that calculates the size into a function and write tests if possible.